src/ivc/cyclefold/incrementally_verifiable_computation/public_params.rs

```diff
@@ -24,22 +24,22 @@ use crate::{
     table::CircuitRunner,
 };
 
-pub struct PublicParams<const A1: usize, const A2: usize, C1, C2, SC>
+pub struct PublicParams<const A1: usize, const A2: usize, CMain, CSup, SC>
 where
-    C1: CurveAffine<Base = <C2 as PrimeCurveAffine>::Scalar>,
-    C2: CurveAffine<Base = <C1 as PrimeCurveAffine>::Scalar>,
-    SC: StepCircuit<A1, C1::Scalar>,
-    C1::Scalar: PrimeFieldBits + FromUniformBytes<64>,
-    C2::Scalar: PrimeFieldBits + FromUniformBytes<64>,
+    CMain: CurveAffine<Base = <CSup as PrimeCurveAffine>::Scalar>,
+    CSup: CurveAffine<Base = <CMain as PrimeCurveAffine>::Scalar>,
+    SC: StepCircuit<A1, CMain::Scalar>,
+    CMain::Scalar: PrimeFieldBits + FromUniformBytes<64>,
+    CSup::Scalar: PrimeFieldBits + FromUniformBytes<64>,
 {
-    ck1: CommitmentKey<C1>,
-    primary_S: PlonkStructure<C1::ScalarExt>,
-    primary_k_table_size: u32,
-    initial_primary_trace: PlonkTrace<C1>,
+    pub primary_ck: CommitmentKey<CMain>,
+    pub primary_S: PlonkStructure<CMain::ScalarExt>,
+    pub primary_k_table_size: u32,
+    pub primary_initial_trace: PlonkTrace<CMain>,
 
-    ck2: CommitmentKey<C2>,
-    secondary_S: PlonkStructure<C2::ScalarExt>,
-    initial_secondary_trace: FoldablePlonkTrace<C2>,
+    pub support_ck: CommitmentKey<CSup>,
+    pub support_S: PlonkStructure<CSup::ScalarExt>,
+    pub support_initial_trace: FoldablePlonkTrace<CSup>,
```

Comment:
what are difference between FoldablePlonkTrace and PlonkTrace? Can we just use one type of trace?